README.md
<img src="https://assets.serlo.org/meta/logo.png" alt="Serlo logo" title="Serlo" align="right" height="60" />

## 🚨 Moved to [`serlo/serlo.org/packages/public/client`](https://github.com/serlo/serlo.org/tree/master/packages/public/client) 🚨

The [`serlo/athene2-assets`](https://github.com/serlo/athene2-assets) repository has been migrated to [`packages/public/client`](https://github.com/serlo/serlo.org/tree/master/packages/public/client) folder of the [`serlo/serlo.org`](https://github.com/serlo/serlo.org) *monorepo*.

We will merge open issues over time and archive this repository afterwards. New issues and pull requests should be created in [`serlo/serlo.org`](https://github.com/serlo/serlo.org).


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

## [4.1.3](https://github.com/serlo/athene2-assets/compare/v4.1.2..v4.1.3) - June 12, 2019

### Fixed

- Don't throw error if `MathJax` is not defined (again)

## [4.1.2](https://github.com/serlo/athene2-assets/compare/v4.1.1..v4.1.2) - June 12, 2019

### Fixed

- Don't throw error if `MathJax` is not defined (again)

## [4.1.1](https://github.com/serlo/athene2-assets/compare/v4.1.0..v4.1.1) - June 5, 2019

### Fixed

- Don't throw error if `MathJax` is not defined

### Internal

- Make sourcemap urls relative to workspace root
- Track issues only from \*.serlo.org

## [4.1.0](https://github.com/serlo/athene2-assets/compare/v4.0.1..v4.1.0) - June 4, 2019

### Changed

- Update dependencies

## [4.0.1](https://github.com/serlo/athene2-assets/compare/v4.0.0..v4.0.1) - June 4, 2019

### Fixed

- Show datetimes correctly

## [4.0.0](https://github.com/serlo/athene2-assets/compare/3.7.3..v4.0.0) - May 29, 2019

### Breaking Changes

- Stylesheet changes

### Changed

- Updated logos of our partners

### Fixed

- Fix image light box on mobile

### Internal

- Build docker image serlo/legacy-editor-renderer

## [3.7.3](https://github.com/serlo/athene2-assets/compare/3.7.2..3.7.3) - February 20, 2019

### Internal

- Automatically deploy assets

## [3.7.2](https://github.com/serlo/athene2-assets/compare/3.7.1..3.7.2) - February 18, 2019

### Fixed

- Fixed footer styles

## [3.7.1](https://github.com/serlo/athene2-assets/compare/3.7.0..3.7.1) - February 12, 2019

### Fixed

- Fixed image upload in new editor

## [3.7.0](https://github.com/serlo/athene2-assets/compare/3.6.0..3.7.0) - February 2, 2019

### Changed

- Improved content styles

### Internal

- Move to CircleCI from Travis
- Use [@splish-me/changelog](https://github.com/splish/changelog) to generate changelogs

## [3.6.0](https://github.com/serlo/athene2-assets/compare/3.5.0..3.6.0) - January 22, 2019

### Added

- Content API: notify parent window of actual content dimensions
- Editor: shortcuts for undo/redo

### Fixed

- Editor: various fixes

## [3.5.0](https://github.com/serlo/athene2-assets/compare/3.4.0..3.5.0) - January 16, 2019

### Added

- H5P.com editor plugin

### Internal

- Upload `yarn.lock` with Google Cloud Function for consistent dependencies

## [3.4.0](https://github.com/serlo/athene2-assets/compare/3.3.0..3.4.0) - January 14, 2019

### Added

- Redirect from <https://start.serlo.org> to team overview document

### Changed

- Update `@serlo/editor` to 0.4.0

### Fixed

- Correct heading styles in new editor
- Set max file size limit of old image upload to 2MB
- Handle rewriting of <https://assets.serlo.org/meta> links correctly

### Internal

- Lazy load ReCAPTCHA

## [3.3.0](https://github.com/serlo/athene2-assets/compare/3.2.2..3.3.0) - December 2, 2018

### Changed

- Handle legacy GeoGebra applets more gracefully

### Fixed

- Fix text plugin

## [3.2.2](https://github.com/serlo/athene2-assets/compare/3.2.1..3.2.2) - December 2, 2018

### Internal

- Associate commit with Sentry release on deployment correctly

## [3.2.1](https://github.com/serlo/athene2-assets/compare/3.2.0..3.2.1) - December 2, 2018

### Fixed

- Always show consent banner on top

### Internal

- Associate commit with Sentry release on deployment

## [3.2.0](https://github.com/serlo/athene2-assets/compare/3.1.1..3.2.0) - December 1, 2018

### Added

- Add consent banner for privacy policy and terms of use (de.serlo.org only for now)

### Internal

- Add Sentry to monitor JavaScript runtime errors

## [3.1.1](https://github.com/serlo/athene2-assets/compare/3.1.0..3.1.1) - November 22, 2018

### Fixed

- Handle `ggt/{id}` links correctly

## [3.1.0](https://github.com/serlo/athene2-assets/compare/3.0.1..3.1.0) - November 18, 2018

### Added

- Style semantic elements in taxonomy

### Fixed

- Polyfill css class `.fa-1-5-x`

## [3.0.1](https://github.com/serlo/athene2-assets/compare/3.0.0..3.0.1) - October 31, 2018

### Changed

- Remove explicit target on `@babel/preset-env` so that [Browserlist's](https://github.com/browserslist/browserslist) recommended config is used

### Fixed

- Replace usages of `String.startsWith` for IE compatibility

## [3.0.0](https://github.com/serlo/athene2-assets/compare/2.0.4..3.0.0) - October 30, 2018

### Breaking Changes

- Removed `CommonsChunkPlugin`, i.e. there are no more `commons.js` and `common.css`
- Renamed `editor` to `legacy-editor`, i.e. one has to include `legacy-editor.js` resp. `legacy-editor.css` instead of `editor.js` resp. `editor.css`
- Asset bundle moved to `https://packages.serlo.org/athene2-assets@a/` resp. `https://packages.serlo.org/athene2-assets@b/` (blue-green deployment)
- Google Cloud Function moved to `https://europe-west1-serlo-assets.cloudfunctions.net/editor-renderer-a` resp. `https://europe-west1-serlo-assets.cloudfunctions.net/editor-renderer-b` (blue-green deployment)
- Needs an element with id `ory-editor-meta-data-wrapper` around the element with id `ory-editor-meta-data`

### Fixed

- Load all plugins in render server for new editor
- When server side rendering fails, still pass state down to client for improved debugging
- Hide double sidebar outside the editor

### Internal

- Dynamically import the new editor and its renderer. This leads to a much smaller bundle size for the initial load.
- Automated blue-green deployment of athene2-assets and server side renderer for new editor
- Automated deployment of Cloudflare workers
- Log version to browser console
- Add a content hash of length 8 to file names of chunks and assets (e.g. fonts and images)
- Upgrade to Webpack 4

## [2.0.4](https://github.com/serlo/athene2-assets/compare/bc6106e006b1633f5e6c15f6af2eef0443d8e81f..2.0.4) - September 28, 2018


__tests__/cloudflare-workers/index.js
import { handleRequest } from '../../src/cloudflare-workers'

expect.extend({
  toHaveBeenCalledWithRequest (fetch, req) {
    try {
      expect(fetch).toHaveBeenCalledTimes(1)

      const [arg1, arg2] = fetch.mock.calls[0]

      if (typeof arg1 === 'string') {
        expect({ ...arg2, url: arg1 }).toEqual(req)
      } else {
        expect(arg1).toEqual(req)
      }

      return {
        pass: true
      }
    } catch (err) {
      return {
        pass: false,
        message: () => err.message
      }
    }
  }
})

describe('Cloudflare Workers', () => {
  let fetch

  beforeEach(() => {
    fetch = jest.fn((...args) => {
      return true
    })
    window.fetch = fetch
  })

  describe('semantic file names for assets', () => {
    it('does rewrite semantic file names', async () => {
      await handleRequest({
        url:
          'https://assets.serlo.org/58eb97b7e5376_7d4211710d8bab642798e39e07788e6f2912e86a/door-handle.gif'
      })

      expect(fetch).toHaveBeenCalledWithRequest({
        url:
          'https://assets.serlo.org/58eb97b7e5376_7d4211710d8bab642798e39e07788e6f2912e86a.gif'
      })
    })

    it('does rewrite semantic file names (legacy)', async () => {
      await handleRequest({
        url:
          'https://assets.serlo.org/legacy/58eb97b7e5376_7d4211710d8bab642798e39e07788e6f2912e86a/door-handle.gif'
      })

      expect(fetch).toHaveBeenCalledWithRequest({
        url:
          'https://assets.serlo.org/legacy/58eb97b7e5376_7d4211710d8bab642798e39e07788e6f2912e86a.gif'
      })
    })

    it(`doesn't rewrite non-semantic file names`, async () => {
      await handleRequest({
        url:
          'https://assets.serlo.org/58eb97b7e5376_7d4211710d8bab642798e39e07788e6f2912e86a.gif'
      })

      expect(fetch).toHaveBeenCalledWithRequest({
        url:
          'https://assets.serlo.org/58eb97b7e5376_7d4211710d8bab642798e39e07788e6f2912e86a.gif'
      })
    })

    it(`doesn't rewrite non-semantic file names (legacy)`, async () => {
      await handleRequest({
        url:
          'https://assets.serlo.org/legacy/58eb97b7e5376_7d4211710d8bab642798e39e07788e6f2912e86a.gif'
      })

      expect(fetch).toHaveBeenCalledWithRequest({
        url:
          'https://assets.serlo.org/legacy/58eb97b7e5376_7d4211710d8bab642798e39e07788e6f2912e86a.gif'
      })
    })
  })
})


scripts/changelog.ts
import { generateChangelog } from '@splish-me/changelog'
import * as fs from 'fs'
import * as path from 'path'
import * as util from 'util'

const releases: Parameters<typeof generateChangelog>['0'] = [
  {
    tagName: '2.0.4',
    date: '2018-09-28'
  },
  {
    tagName: '3.0.0',
    date: '2018-10-30',
    breakingChanges: [
      'Removed `CommonsChunkPlugin`, i.e. there are no more `commons.js` and `common.css`',
      'Renamed `editor` to `legacy-editor`, i.e. one has to include `legacy-editor.js` resp. `legacy-editor.css` instead of `editor.js` resp. `editor.css`',
      'Asset bundle moved to `https://packages.serlo.org/athene2-assets@a/` resp. `https://packages.serlo.org/athene2-assets@b/` (blue-green deployment)',
      'Google Cloud Function moved to `https://europe-west1-serlo-assets.cloudfunctions.net/editor-renderer-a` resp. `https://europe-west1-serlo-assets.cloudfunctions.net/editor-renderer-b` (blue-green deployment)',
      'Needs an element with id `ory-editor-meta-data-wrapper` around the element with id `ory-editor-meta-data`'
    ],
    fixed: [
      'Load all plugins in render server for new editor',
      'When server side rendering fails, still pass state down to client for improved debugging',
      'Hide double sidebar outside the editor'
    ],
    internal: [
      'Dynamically import the new editor and its renderer. This leads to a much smaller bundle size for the initial load.',
      'Automated blue-green deployment of athene2-assets and server side renderer for new editor',
      'Automated deployment of Cloudflare workers',
      'Log version to browser console',
      'Add a content hash of length 8 to file names of chunks and assets (e.g. fonts and images)',
      'Upgrade to Webpack 4'
    ]
  },
  {
    tagName: '3.0.1',
    date: '2018-10-31',
    changed: [
      "Remove explicit target on `@babel/preset-env` so that [Browserlist's](https://github.com/browserslist/browserslist) recommended config is used"
    ],
    fixed: ['Replace usages of `String.startsWith` for IE compatibility']
  },
  {
    tagName: '3.1.0',
    date: '2018-11-18',
    added: ['Style semantic elements in taxonomy'],
    fixed: ['Polyfill css class `.fa-1-5-x`']
  },
  {
    tagName: '3.1.1',
    date: '2018-11-22',
    fixed: ['Handle `ggt/{id}` links correctly']
  },
  {
    tagName: '3.2.0',
    date: '2018-12-01',
    added: [
      'Add consent banner for privacy policy and terms of use (de.serlo.org only for now)'
    ],
    internal: ['Add Sentry to monitor JavaScript runtime errors']
  },
  {
    tagName: '3.2.1',
    date: '2018-12-02',
    fixed: ['Always show consent banner on top'],
    internal: ['Associate commit with Sentry release on deployment']
  },
  {
    tagName: '3.2.2',
    date: '2018-12-02',
    internal: ['Associate commit with Sentry release on deployment correctly']
  },
  {
    tagName: '3.3.0',
    date: '2018-12-02',
    changed: ['Handle legacy GeoGebra applets more gracefully'],
    fixed: ['Fix text plugin']
  },
  {
    tagName: '3.4.0',
    date: '2019-01-14',
    added: ['Redirect from https://start.serlo.org to team overview document'],
    changed: ['Update `@serlo/editor` to 0.4.0'],
    fixed: [
      'Correct heading styles in new editor',
      'Set max file size limit of old image upload to 2MB',
      'Handle rewriting of https://assets.serlo.org/meta links correctly'
    ],
    internal: ['Lazy load ReCAPTCHA']
  },
  {
    tagName: '3.5.0',
    date: '2019-01-16',
    added: ['H5P.com editor plugin'],
    internal: [
      'Upload `yarn.lock` with Google Cloud Function for consistent dependencies'
    ]
  },
  {
    tagName: '3.6.0',
    date: '2019-01-22',
    added: [
      'Content API: notify parent window of actual content dimensions',
      'Editor: shortcuts for undo/redo'
    ],
    fixed: ['Editor: various fixes']
  },
  {
    tagName: '3.7.0',
    date: '2019-02-02',
    changed: ['Improved content styles'],
    internal: [
      'Move to CircleCI from Travis',
      'Use [@splish-me/changelog](https://github.com/splish/changelog) to generate changelogs'
    ]
  },
  {
    tagName: '3.7.1',
    date: '2019-02-12',
    fixed: ['Fixed image upload in new editor']
  },
  {
    tagName: '3.7.2',
    date: '2019-02-18',
    fixed: ['Fixed footer styles']
  },
  {
    tagName: '3.7.3',
    date: '2019-02-20',
    internal: ['Automatically deploy assets']
  },
  {
    tagName: 'v4.0.0',
    name: '4.0.0',
    date: '2019-05-29',
    breakingChanges: ['Stylesheet changes'],
    changed: ['Updated logos of our partners'],
    fixed: ['Fix image light box on mobile'],
    internal: ['Build docker image serlo/legacy-editor-renderer']
  },
  {
    tagName: 'v4.0.1',
    name: '4.0.1',
    date: '2019-06-04',
    fixed: ['Show datetimes correctly']
  },
  {
    tagName: 'v4.1.0',
    name: '4.1.0',
    date: '2019-06-04',
    changed: ['Update dependencies']
  },
  {
    tagName: 'v4.1.1',
    name: '4.1.1',
    date: '2019-06-05',
    fixed: ["Don't throw error if `MathJax` is not defined"],
    internal: [
      'Make sourcemap urls relative to workspace root',
      'Track issues only from *.serlo.org'
    ]
  },
  {
    tagName: 'v4.1.2',
    name: '4.1.2',
    date: '2019-06-12',
    fixed: ["Don't throw error if `MathJax` is not defined (again)"]
  },
  {
    tagName: 'v4.1.3',
    name: '4.1.3',
    date: '2019-06-12',
    fixed: ["Don't throw error if `MathJax` is not defined (again)"]
  }
]

const writeFile = util.promisify(fs.writeFile)

exec()

async function exec(): Promise<void> {
  const content = await generateChangelog(releases)

  await writeFile(path.join(__dirname, '..', 'CHANGELOG.md'), content)
}
